api/db-cdr/generate-cdr/libproject/devstart.rb

```diff
@@ -59,6 +59,14 @@ def bq_ingest(tier, source_project, dataset_name, table_match_filter="", table_s
   # bq init --delete_credentials as recommended in the output.
   # TODO(RW-3768): Find a better solution for Google credentials in docker.
 
+  # validate the CDR's tier label against the user-supplied tier, as a safety check
+
+  cdr_metadata = JSON.parse(`bq show --format=json "#{source_fq_dataset}"`)
+  dataset_tier = cdr_metadata['labels']['data_tier']
```

Comment:
If "data_tier" is missing entirely, logging a warning may be appropriate instead. I can imagine this might be the case in the event that we're publishing a synthetic dataset or lower environment.

Reply:
yes, that is what we see on test.  updated.